manager/src/main/java/com/redhat/service/smartevents/manager/BridgesServiceImpl.java

```diff
@@ -90,6 +91,53 @@ public Bridge createBridge(String customerId, String organisationId, String owne
         return bridge;
     }
 
+    @Override
+    @Transactional
+    public Bridge updateBridge(String bridgeId, String customerId, BridgeRequest bridgeRequest) {
+        // Extract existing definition
+        Bridge existingBridge = getBridge(bridgeId, customerId);
+
+        if (!existingBridge.isActionable()) {
+            throw new BridgeLifecycleException(String.format("Bridge with id '%s' for customer '%s' is not in an actionable state.",
+                    bridgeId,
+                    customerId));
+        }
+
+        BridgeDefinition existingDefinition = existingBridge.getDefinition();
+
+        // Validate update.
+        // Name cannot be updated.
+        if (!Objects.equals(existingBridge.getName(), bridgeRequest.getName())) {
+            throw new BadRequestException("It is not possible to update the Bridge's name.");
+        }
+
+        // Construct updated definition
+        Action updatedErrorHandler = bridgeRequest.getErrorHandler();
+        BridgeDefinition updatedDefinition = new BridgeDefinition(updatedErrorHandler);
+
+        // No need to update CRD if the definition is unchanged
+        if (existingDefinition.equals(updatedDefinition)) {
```

Comment:
Could we please use `Objects.equals()` ? reading this class, not sure `existingDefinition` is `!= null`.

Reply:
Done. I forgot your fondness of `Objects.equals(..)` (and I do not disagree, it is a good idea).